src/dodal/devices/i04/transfocator.py

```diff
@@ -0,0 +1,58 @@
+from time import sleep
+
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsSignal, EpicsSignalRO, Kind
+from ophyd.status import DeviceStatus
+
+from dodal.log import LOGGER
+
+
+class Transfocator(Device):
+    beamsize_set: EpicsSignal = Cpt(EpicsSignal, "VERT_REQ", kind=Kind.hinted)
+    predict_vertical: EpicsSignal = Cpt(EpicsSignal, "LENS_PRED")
+
+    number_filters_sp: EpicsSignal = Cpt(EpicsSignal, "NUM_FILTERS")
+
+    start: EpicsSignal = Cpt(EpicsSignal, "START.PROC")
+    start_rbv: EpicsSignalRO = Cpt(EpicsSignalRO, "START_RBV")
+
+    vertical_lens_rbv: EpicsSignalRO = Cpt(EpicsSignalRO, "VER", kind=Kind.hinted)
+
+    TIMEOUT = 120
+    _POLLING_WAIT = 0.01
+
+    def polling_wait_on_stat_rbv(self, for_value):
+        # For some reason couldn't get monitors working on START_RBV
+        # (See https://github.com/DiamondLightSource/dodal/issues/152)
+        for _ in range(int(self.TIMEOUT / self._POLLING_WAIT)):
+            RBV_value = self.start_rbv.get()
+            sleep(self._POLLING_WAIT)
+            if RBV_value == for_value:
+                return
+        raise TimeoutError()
```

Comment:
This might take much longer than `TIMEOUT` seconds, especially if `.get()` is slow for any reason.

How about something like:

```python
def polling_wait_on_stat_rbv(self, for_value):
    start_time = time.time()
    while time.time() < start_time + self.TIMEOUT:
        RBV_value = self.start_rbv.get()
        if RBV_value == for_value:
                return
        sleep(self._POLLING_WAIT)

    # last try
    if self.start_rbv.get() != for_value:
        raise TimeoutError()
```